src/Bonsai.ML.Torch/OpenCVHelper.cs

```diff
@@ -79,24 +29,21 @@ public static unsafe Tensor CreateTensorFromPtr(IntPtr tensorDataPtr, long[] dim
         public static Tensor ToTensor(IplImage image)
         {
             if (image == null)
-            {
                 return empty([ 0, 0, 0 ]);
-            }
-            // int width = image.Width;
+
             int height = image.Height;
             int channels = image.Channels;
             var width = image.WidthStep / channels;
 
             var iplDepth = image.Depth;
             var tensorType = bitDepthLookup.FirstOrDefault(x => x.Value.IplDepth == iplDepth).Key;
 
-            IntPtr tensorDataPtr = image.ImageData;
-            long[] dimensions = [ height, width, channels ];
-            if (tensorDataPtr == IntPtr.Zero) 
-            {
-                return empty(dimensions);
-            }
-            return CreateTensorFromPtr(tensorDataPtr, dimensions, tensorType);
+            IntPtr data = image.ImageData;
+            ReadOnlySpan<long> dimensions = stackalloc long[] { height, width, channels };
+            if (data == IntPtr.Zero) 
+                return zeros(dimensions);
```

Comment:
I changed these from `empty` because it seems wrong to return an uninitialized tensor in this situation, but I later learned that apparently zero tensors are immutable by default? So maybe this isn't appropriate.

I'd also consider whether this is the appropriate thing to do here at all though. Are there any normal situations where a non-null `IplImage`/`Mat` has no backing memory like this? Seems like it should throw an exception.

Reply:
Talked to Nick about this on Teams and there isn't a known situation where this is known to happen so changed it to an exception instead. [`UnreachableException`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.unreachableexception?view=net-8.0) would've been ideal since we don't think this situation is possible, but we have to support .NET Framework.